docs/DesignDocs/SGX_EMM.md

```diff
@@ -0,0 +1,678 @@
+SGX Enclave Memory Manager
+=================================
+
+## Motivation ##
+
+An enclave's memory is backed by a special reserved region in RAM, called
+Enclave Page Cache (EPC). Enclave memory management tasks include
+allocating/reserving virtual address ranges, committing physical EPC pages,
+changing EPC page permissions or page types, and removing EPC pages.
+Those tasks require collaboration between the trusted runtime, the untrusted
+runtime, and the OS. The SGX enclave memory manager (EMM) serves as a central
+component in the enclave trusted runtime that abstracts the interaction with
+the untrusted runtime for all memory management flows and provides APIs for
+its clients to reserve virtual address ranges, commit EPC memory to the reserved
+address ranges, and modify attributes of the reserved/committed pages.
+
+For details of specific memory management related flows, please refer to
+[the SGX EDMM driver API spec](https://github.com/openenclave/openenclave/docs/DesignDocs/SGX2-EDMM-driver-interface.md).
+The public EMM APIs defined here are most likely invoked by some intermediate
+runtime level components for specific usages, such as dynamic heap/stack, mmap,
+mprotect, higher level language JIT compiler, etc.
+
+
+## User Experience ##
+
+**Allocate, Deallocate Enclave Memory**
+
+The EMM provides an API, sgx_mm_alloc, for clients to request enclave memory
```

Comment:
Clarify: when this says "the EMM provides an API", which SDK/library is this referring to?  PSW?  The Intel SGX SDK for Linux?  The Intel SGX SDK for Windows?  (Presumably it's not the OE SDK or the API name would start with oe_ instead.)  And what is the "client" here, is it an app or the OE SDK?  (I *think* this is answered down at line 78 but since it's way below, this text is confusing.  I might suggest moving the Runtime Abstraction Layer section starting at 72 up before this section.)

Reply:
will move up the abstraction layer section to give more context. thx